blockchain/blockchain.go

```diff
@@ -40,23 +38,21 @@ type Reader interface {
 	HeadState() (core.StateReader, StateCloser, error)
 	StateAtBlockHash(blockHash *felt.Felt) (core.StateReader, StateCloser, error)
 	StateAtBlockNumber(blockNumber uint64) (core.StateReader, StateCloser, error)
-	PendingState() (core.StateReader, StateCloser, error)
 
 	BlockCommitmentsByNumber(blockNumber uint64) (*core.BlockCommitments, error)
 
 	EventFilter(from *felt.Felt, keys [][]felt.Felt) (EventFilterer, error)
 
-	Pending() (Pending, error)
-
 	Network() *utils.Network
 }
 
 var (
 	ErrParentDoesNotMatchHead = errors.New("block's parent hash does not match head block hash")
+	ErrPendingBlockNotFound   = errors.New("pending block not found")
 	SupportedStarknetVersion  = semver.MustParse("0.13.3")
 )
 
-func checkBlockVersion(protocolVersion string) error {
+func CheckBlockVersion(protocolVersion string) error {
```

Comment:
So you making this function public because previously it was used only in `blockchain`, but now since `blockchain` is not responsible for storing pending block you need to call it in `sync` (because pending block is stored there). Correct ?